### PR TITLE
Amend CI testing to not build on arm, correctly cross compile on arm64.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-arch: ['amd64', 'arm', 'arm64']
+        go-arch: ['amd64', 'arm64']
```
Comment (DimitrisJim) [inline .github/workflows/test.yml:39]: Reasoning: https://github.com/CosmWasm/wasmvm#builds-of-libwasmvm

> Our system currently supports the following builds. In general we can only support targets that are [supported by Wasmer's singlepass backend](https://docs.wasmer.io/ecosystem/wasmer/wasmer-features#compiler-support-by-chipset), which for example excludes all 32 bit systems.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -49,15 +49,22 @@ jobs:
...
+      # Install cross compiler for ARM64. Export CC env variable.
+      - name: Install compiler for arm64.
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc-aarch64-linux-gnu
+          echo "CC=aarch64-linux-gnu-gcc" >> $GITHUB_ENV
+        if: matrix.go-arch == 'arm64'
       - name: Build ibc-go
-        run: GOARCH=${{ matrix.go-arch }} LEDGER_ENABLED=false make build
+        run: GOARCH=${{ matrix.go-arch }} CGO_ENABLED=1 LEDGER_ENABLED=false make build
```
Comment (DimitrisJim) [inline .github/workflows/test.yml:60]: `CGO_ENABLED` is, by default, disabled if targeting an arch different than native.

**Comment by DimitrisJim** [comment]
this could also target main? Lmk what people think.

**Review by charleenfei — APPROVED** [review]

**Review by crodriguezvega — APPROVED** [review]
Great work figuring this out, @DimitrisJim and @charleenfei!

**Pull request merged**